_examples/postcard/storage/simple_postgres_eventstore.go

```diff
@@ -84,6 +84,24 @@ func NewSimpleSQLitePostcardRepository(ctx context.Context, db *sql.DB) (eventst
 	)
 }
 
+func NewGOBSQLitePostcardRepository(ctx context.Context, db *sql.DB) (eventstore.EventStore[postcard.Postcard], error) {
+	return eventstore.NewSQLStore[postcard.Postcard](
+		ctx,
+		db,
+		eventstore.SQLConfig[postcard.Postcard]{
+			SchemaAdapter: eventstore.NewSQLiteSchemaAdapter[postcard.Postcard](""),
+			Serializer: transport.NewGOBSerializer[postcard.Postcard](
+				[]stream.Event[postcard.Postcard]{
+					postcard.Created{},
+					postcard.Addressed{},
+					postcard.Written{},
+					postcard.Sent{},
+				},
+			),
```

Comment:
I think you could achieve the same with:

```go
type GobMarshaler struct{}

func (GobMarshaler) Marshal(_ stream.ID, data interface{}) ([]byte, error) {
	// ...
}

func (GobMarshaler) Unmarshal(_ stream.ID, bytes []byte, target interface{}) error {
	// ...
}
```

And then:

```go
			Serializer: transport.NewSimpleSerializer[postcard.Postcard](
				GobMarshaler{},
 				[]stream.Event[postcard.Postcard]{
 					postcard.Created{},
 					postcard.Addressed{},
 					postcard.Written{},
 					postcard.Sent{},
 				},
 			),
```

The point is that marshaling in this context is converting a struct to a slice of bytes and vice versa, which is very straightforward.

The naming could be improved probably, so it's more clear.

Reply:
oh yes, makes perfect sense! And it will be much simpler. I will try it tomorrow. Thanks!

Reply:
https://github.com/ThreeDotsLabs/esja/pull/17/commits/98ebf5dde8fde9f2417f954c8df41ad0613df579